### PR TITLE
Map detect location

### PR DESCRIPTION
**Pull request by RobertoCarrilloAvila**
## Screenshots:
Screenshot 2024-01-25 at 9 26 31 p m

### CODE REVIEW

File: src/components/Map/Map.js [hunk 1]
```diff
@@ -1,24 +1,46 @@
-import { useState, useEffect } from "react";
-import { createMap, setCurrentLocationAsCenter } from "services/MapService";
+import { useState, useEffect } from 'react';
+import { GoogleMap, useLoadScript } from '@react-google-maps/api';
 import "components/Map/Map.scss";
 
-const MAP_ID = "map";
+const defaultLocation = { lat: 19.432, lng: -99.133 };
+const defaultZoom = 12;
```
Comment (flomosq) [inline src/components/Map/Map.js:6]: Constants should be named in all caps
```suggestion
const DEFAULT_LOCATION = { lat: 19.432, lng: -99.133 };
const DEFAULT_ZOOM = 12;
```
And I also think creating a constants file near the Map component might be a useful place to put them in

File: src/components/Map/Map.js [hunk 2]
```diff
@@ -1,24 +1,46 @@
...
-
-    initializeMap();
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition((position) => {
+        setCurrentLocation({
+          lat: position.coords.latitude,
+          lng: position.coords.longitude
+        });
+      });
+    }
```
Comment (flomosq) [inline src/components/Map/Map.js:23]: Looks a bit cleaner IMO
```suggestion
    navigator.geolocation?.getCurrentPosition(({ coords }) => {
      setCurrentLocation({
        lat: coords.latitude,
        lng: coords.longitude
      });
    });
```

File: src/components/Map/Map.js [hunk 3]
```diff
@@ -1,24 +1,46 @@
...
   return (
-    <div id={ MAP_ID }></div>
-  );
+    <div className="map">
+      <GoogleMap
+        mapContainerClassName="map__container"
+        center={currentLocation}
+        zoom={defaultZoom} >
+
+      </GoogleMap>
```
Comment (flomosq) [inline src/components/Map/Map.js:41]:
```suggestion
      <GoogleMap
        mapContainerClassName="map__container"
        center={currentLocation}
        zoom={defaultZoom}
      />
```

File: src/components/Map/Map.js [hunk 4]
```diff
@@ -1,24 +1,46 @@
-import { useState, useEffect } from "react";
-import { createMap, setCurrentLocationAsCenter } from "services/MapService";
+import { useState, useEffect } from 'react';
+import { GoogleMap, useLoadScript } from '@react-google-maps/api';
 import "components/Map/Map.scss";
```
Comment (afmicc) [inline src/components/Map/Map.js:3]:
```suggestion
import 'components/Map/Map.scss';
```
you should be able to configure this with eslint

Comment (RobertoCarrilloAvila) [reply src/components/Map/Map.js:3]: I'll fix them and I'll add the configuration in [this PR](https://github.com/RobertoCarrilloAvila/target/pull/19)

**Pull request merged**